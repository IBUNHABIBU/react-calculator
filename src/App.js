import logo from './logo.svg';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} className="App-logo" alt="logo" />
        <p>
          <code>React calculator</code>
        </p>
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Fully functional react calculator app
        </a>
      </header>
    </div>
  );
}

export default App;
