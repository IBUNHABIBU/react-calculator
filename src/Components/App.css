:root {
  --primary-color:  rgb(128, 128, 128);
  --primay-height: 100px;
  --primary-width: 700px;
}

.app-container {
  width: var(--primary-width);
}

.display-container {
  display: flex;
  height: var(--primay-height);
  color: #fff;
  background-color: var(--primary-color);
  padding: 4px;
  font-weight: bold;
  justify-content: flex-end;
  align-items: center;
}

.btn-panel {
  display: flex;
  height: 100px;
}

.btn {
  width: calc( var(--primary-width) * 0.25);
  font-size: 1.4rem;
  font-weight: bold;
}

.btn.zero {
  width: calc( var(--primary-width) * 0.5 );
}

.btn.o-class {
  background-color: #ff9000;
}
