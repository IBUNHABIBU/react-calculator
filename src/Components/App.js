/* eslint-disable no-unused-vars */
import React, { Component } from 'react';
import Display from './Display';
import ButtonPanel from './ButtonPanel';
import calculate from '../logic/calculate';

export default class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '0',
    };
  }

  handleClick = (btnName) => {
    const newState = this.state;
    this.setState((data) => calculate(data, btnName));
  }

  render() {
    const { total } = this.state;
    return (
      <>
        <Display result= {total} />
        <ButtonPanel clickHandler={this.handleClick} />
      </>
    );
  }
}
