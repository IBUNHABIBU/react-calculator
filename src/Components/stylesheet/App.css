:root {
  --primary-color: rgb(128, 128, 128);
  --primay-height: 100px;
  --primary-width: 700px;
}

.main-container {
  display: flex;
  justify-content: flex-end;
  padding-right: 2rem;
}

.app-container {
  width: var(--primary-width);
}

.display-container {
  display: flex;
  height: var(--primay-height);
  color: #fff;
  background-color: var(--primary-color);
  padding: 4px;
  font-size: 2rem;
  font-weight: bold;
  justify-content: flex-end;
  align-items: center;
}

.btn-panel {
  display: flex;
  height: 80px;
  justify-content: flex-end;
}

.btn {
  width: calc(var(--primary-width) * 0.25);
  font-size: 1rem;
  font-weight: bold;
  border: 1px solid var(--primary-color);
}

.btn.zero {
  width: calc(var(--primary-width) * 0.5);
}

.btn.o-class {
  background-color: #ff9000;
}
