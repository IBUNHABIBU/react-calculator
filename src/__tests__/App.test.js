/* globals describe, expect, it, beforeEach */
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import App from '../Components/App';

let wrapper;
beforeEach(() => {
  wrapper = shallow(<App />);
});

describe('Combine all components', () => {
  it('should have Display component', () => {
    expect(wrapper.find('.app-container').text()).toContain('<Display />');
  });

  it('should have ButtonPanel component', () => {
    expect(wrapper.find('.app-container').text()).toContain('<ButtonPanel />');
  });
});

describe("snapshots", () => {
  it("renders App component correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  })
})
