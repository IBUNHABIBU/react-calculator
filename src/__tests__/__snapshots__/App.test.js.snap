// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots renders App component correctly 1`] = `
<div
  className="main-container"
>
  <div
    className="app-container"
  >
    <div
      className="display-container"
    >
        
    </div>
    <div
      className="btn-panel"
    >
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="o-class  btn"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
    </div>
    <div
      className="btn-panel"
    >
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="o-class  btn"
        onClick={[Function]}
        type="button"
      >
        X
      </button>
    </div>
    <div
      className="btn-panel"
    >
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="o-class  btn"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
    </div>
    <div
      className="btn-panel"
    >
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="o-class  btn"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
    </div>
    <div
      className="btn-panel"
    >
      <button
        className=" zero btn"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="  btn"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="o-class  btn"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
